### PR TITLE
Playlist

### PR DESCRIPTION
**Pull request by alex-schechter**

### CODE REVIEW

File: src/app/playlist/playlist-edit/playlist-edit.component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  
+  //TODO add the singed-in user as song for this playlist - and disable his delete option 
+  //Temp array for the selected songs, send to server only on submit
+  selected_songs: Song[];
+
+
+  //TODO - remove the hard-coded songs and send real db queries as explained in the html - leave the list as empty list
+  songs: string[]= ['CardiB','Catey','CICI','Pink','Pupi','Marshmelo','Melo','Khalid','Kuki','Bruno M'];  
+
```
Comment (DaniSchechter) [inline src/app/playlist/playlist-edit/playlist-edit.component.ts]: If u copy check what u r copying

File: src/app/playlist/playlist-edit/playlist-edit.component.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  // !!!! TODO change to load only one playlist and not the entire playlists
+  ngOnInit() {
+    this.selected_songs = [];
+    this.playlist = this.playlistService.getPlaylist(this.playlists, this.playlistId);
+    this.route.paramMap.subscribe((paramMap: ParamMap) => {
+        if (paramMap.has('id')){
+            this.playlistId = paramMap.get('id');
+            this.playlistService.getPlaylist(this.playlists, this.playlistId);
+            this.playlistService.getPlaylists();
```
Comment (DaniSchechter) [inline src/app/playlist/playlist-edit/playlist-edit.component.ts]: Add comment for each line. All looks the same for me

File: src/app/playlist/playlist-edit/playlist-edit.component.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    });
+  }
+
+  // Adds an song that was selected to playlist's songs list
+  onSelectSong(song: Song) {
+    this.selected_songs.push(song);
+  }
+
+  // delete this song from the selected song list
+  onDeleteSelectedSong(song_to_delete: string) {
```
Comment (DaniSchechter) [inline src/app/playlist/playlist-edit/playlist-edit.component.ts]: String? Not song? How does it work?

File: src/app/playlist/playlist.service.ts [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        }))
+        .subscribe(
+            (playlistsAfterChange) => {
+                this.playlistsCount = playlistsAfterChange.totalPlaylists;
+                this.playlists = playlistsAfterChange.playlists;
+                this.playlistsUpdated.next({
+                    playlists: [...this.playlists], 
+                    totalPlaylists: playlistsAfterChange.totalPlaylists
+                });
+                    //! TODO think if diplay message on fetching
```
Comment (DaniSchechter) [inline src/app/playlist/playlist.service.ts]: Remove

File: src/app/song/song-tool-bar/song-tool-bar.component.ts [hunk 5]
```diff
@@ -45,20 +51,63 @@ export class SongToolBarComponent implements OnInit {
...
+        this.selectedPlaylists = null;
+    })
+  }
+
+
+    //TODO change to real action for the next 2 buttons 
   onPlay() {alert("song "+ this.song.name +" playnow")}
   onAddToQueue() {alert("song "+ this.song.name +" queue")}
   onLikeToggle() { 
+    console.log({"liked":this.songLiked});
```
Comment (DaniSchechter) [inline src/app/song/song-tool-bar/song-tool-bar.component.ts]: Remove

File: src/app/song/song-tool-bar/song-tool-bar.component.ts [hunk 6]
```diff
@@ -45,20 +51,63 @@ export class SongToolBarComponent implements OnInit {
...
+    }
   }
 
   onDelete() {
     this.songsService.deleteSong(this.song.id);
   }
+
+  saveToPlaylists(song: Song){
+    if (!this.selectedPlaylists || this.selectedPlaylists.length == 0){
+      alert("You didnt enter any songs");
```
Comment (DaniSchechter) [inline src/app/song/song-tool-bar/song-tool-bar.component.ts]: Notification

File: src/app/playlist/playlist-edit/playlist-edit.component.ts (hunk 3 above)
Comment (alex-schechter) [reply src/app/playlist/playlist-edit/playlist-edit.component.ts]: id of song

**Pull request merged**